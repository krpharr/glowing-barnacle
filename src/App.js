import React from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";
import Directory from "./pages/Directory";
import EmployeeProfile from "./pages/EmployeeProfile";


function App() {
  return (
    <Router>
      <Route exact path="/" component={Directory} />
      <Route exact path="/employee/:id" component={EmployeeProfile} />
    </Router>
  );
}

export default App;