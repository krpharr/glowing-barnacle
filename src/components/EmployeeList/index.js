import React from "react";

function EmployeeList(props){
    mapEmployees = () =>  {
        let empListItems = props.currentList.map(emp => {
            return (
                <li key={emp.id}> 
                    <img alt={emp.email} src={emp.avatar}></img>
                    <h2>{emp.firstName} {emp.lastName}</h2>
                    <h4>{emp.department} | {emp.title}</h4>
                    <span>{emp.email} {emp.phone}</span>
                </li>
            );
        });
        return empListItems;
    };
    return (
        <ul>
            {this.mapEmployees()}
        </ul>
    );
}

export default EmployeeList;