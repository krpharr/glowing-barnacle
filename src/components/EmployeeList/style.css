.employee-list {
    list-style-type: none;
}