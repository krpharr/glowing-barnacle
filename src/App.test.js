import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import EmployeeList from './components/EmployeeList';

describe('Snapshot test', () => {
  test('App snapshot test', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON;
    expect(tree).toMatchSnapshot();
  });
});

describe('App getByText tests', () => {
  test('renders header title', () => {
    const { getByText } = render(<App />);
    const text = getByText(/Sort Schmort, Inc./i);
    expect(text).toBeInTheDocument();
  });

  test('renders Department:', () => {
    const { getByText } = render(<App />);
    const text = getByText(/Department:/i);
    expect(text).toBeInTheDocument();
  });

  test('get select box', () => {
    const { getByDisplayValue } = render(<App />);
    const select = getByDisplayValue(/Department/i);
    expect(select).toBeInTheDocument();
  });

  test("get div with text = 'Sort'", () => {
    const { getByTestId } = render(<App />);
    const text = getByTestId(/sortTxtId/i);
    expect(text).toBeInTheDocument();
  });
});
