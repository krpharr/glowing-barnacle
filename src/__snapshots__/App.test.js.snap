// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test App snapshot test 1`] = `[Function]`;
